packages/runtime/src/widgets/Collapsible.tsx

```diff
@@ -71,9 +76,86 @@ export type CollapsibleProps = {
 } & EnsembleWidgetProps &
   HasItemTemplate;
 
+const EvaluateItem = React.memo(
+  ({
+    context,
+    template,
+    currentIndex,
+    setEvaluated,
+  }: {
+    context: object;
+    currentIndex: number;
+    template: {
+      key: string;
+      label: string | EnsembleWidget;
+      children: EnsembleWidget;
+    };
+    setEvaluated: React.Dispatch<React.SetStateAction<CollapsibleTemplate[]>>;
+  }): null => {
+    const evaluatedData = useEvaluate({ ...template }, { context });
+
+    useEffect(() => {
+      const { key, label, children } = evaluatedData;
+
+      setEvaluated((prev) => {
+        prev[currentIndex] = {
+          key,
+          label: isString(label) ? label : EnsembleRuntime.render([label]),
+          children: EnsembleRuntime.render([children]),
+        };
+        return [...prev];
+      });
+    }, [evaluatedData]);
+
+    return null;
+  },
+);
+
+EvaluateItem.displayName = "EvaluateItem";
+
+const withValueTemplate = <P extends object>(
```

Comment:
There is no need for this to be a HOC since the logic is not shared anywhere. This is just extra syntax for no value